packages/react-components/src/components/knowledge-graph/responseParser.tsx

```diff
@@ -0,0 +1,66 @@
+import { INodeResults, IRelationResults } from './interfaces';
+import { ExecuteQueryCommandOutput } from '@aws-sdk/client-iottwinmaker';
+import { NodeData, EdgeData } from './graph/types';
+
+function parseNode(item: INodeResults, nodeData: Map<string, NodeData>) {
+  const entityId = item.entityId;
+  const entityName = item.entityName;
+  const entityData: INodeResults = { ...item };
+  nodeData.set(entityId, {
+    entityData,
+    id: entityId,
+    label: entityName ? entityName : '',
+  });
+}
+function parseEdge(item: IRelationResults, edgeData: Map<string, EdgeData>) {
+  const sourceId = item.sourceEntityId;
+  const targetId = item.targetEntityId;
+  const relationshipName = item.relationshipName;
+  const id = `${sourceId}-${targetId}`;
+  edgeData.set(id, {
+    id,
+    label: relationshipName ? relationshipName : '',
+    source: sourceId,
+    target: targetId,
+  });
+}
```

Comment:
I didn't understand what you mean by "basic mapping code that probably doesn't exist." However mapping to unique id is necessary to avoid edge & node duplicates thats the whole point of using Map to represent nodes and edges. Otherwise we have to have logic to check for duplicates and take them away.